### PR TITLE
docs: add a release process document

### PR DESCRIPTION
**Pull request by phlogistonjohn**
This is based on the process followed to create the recent release and should serve as a guide in case someone else does a future release.

### CODE REVIEW

File: docs/release-process.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+> [!IMPORTANT]
+> The projects that the samba-operator includes/depends on must be released
+> *before* the samba-operator. As a policy the "SINK" team releases the projects
+> as a group. See the section below for guidance with regards to version numbers.
+
+For example, if we plan on releasing operator v0.6 we first release
+samba-containers v0.6, and so on. If a project is unchanged we do not need to
+make a release and the older version number will continue to be used.  If we do
+need to make a release that was not part of the previous group  we may *skip*
+versions numers. For example if component X was released as v0.3 in along with
```
Comment (anoopcs9) [inline docs/release-process.md:39]:
```suggestion
versions numbers. For example if component X was released as v0.3 in along with
```

File: docs/release-process.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+samba-operator v0.3 but was unchanged for operator v0.4 we would release it as
+v0.5 along with samba-operator v0.5 if a release is needed for that project.
+
+Once the dependency image tags have been updated, commit the change. For example:
+```
+git commit -s -p -m 'release: use v0.5 tag for samba-server and samba-metrics images'
+```
+
+Run `git push` to push the change to the release branch on GitHub, initiating
+a CI test run. Verify the the release branch build passes the tests.
```
Comment (anoopcs9) [inline docs/release-process.md:49]:
```suggestion
a CI test run. Verify the release branch build passes the tests.
```

File: docs/release-process.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+In the meantime, edit the file `config/manager/manager.yaml` to bump up
+the version of the operator image tag (to match the upcoming tag). This file
+is used when deploying the operaterator and must be configured to deploy
```
Comment (anoopcs9) [inline docs/release-process.md:53]:
```suggestion
is used when deploying the operator and must be configured to deploy
```

File: docs/release-process.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+the intended release version. Commit the change with a cmmand like so:
```
Comment (anoopcs9) [inline docs/release-process.md:54]:
```suggestion
the intended release version. Commit the change with a command like so:
```

File: docs/release-process.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+Once the release notes are drafted and then either immediately before or after
+publishing them, use the quay.io UI to copy each pre-release tag to the "latest"
+tag and a final "vX.Y" tag. Delete the temporary pre-release tags using the
+quay.io UI as they are no longer needed.
+
+
+## Announcements
+
+As the samba-operator is typically the final release is the group of "SINK"
```
Comment (anoopcs9) [inline docs/release-process.md:172]:
```suggestion
As the samba-operator is typically the final release in the group of "SINK"
```

File: docs/release-process.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+projects that are released. Now is a good time to announce the releases.
```
Comment (anoopcs9) [inline docs/release-process.md:173]: Shouldn't we merge the latter with previous sentence?

File: docs/release-process.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+kubectl apply -f samba-operator-v0.5-default.yaml
+
+This is equivalent to checking out the v0.5 tag from the git repository and using the default configuration. Example:
+
+git clone -b v0.5 https://github.com/samba-in-kubernetes/samba-operator
+cd samba-operator
+kubectl apply -k config/default
+
+```
+... using the image that was pushed earlier
```
Comment (anoopcs9) [inline docs/release-process.md:151]: Above sentence is missing some context. Is this the right place?

**Review by anoopcs9 — CHANGES_REQUESTED** [review]

File: docs/release-process.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+# resolve any conflicts
+```
+
+Now we need to update the versions of the containers that the operator deploys
+by default. Edit `internal/conf/config.go` to update the versions of
+the samba-server, samba-metrics, and svcwatch containers if there are new
+released versions for these images.
+
+> [!IMPORTANT]
+> The projects that the samba-operator includes/depends on must be released
```
Comment (obnoxxx) [inline docs/release-process.md:31]: we should name/link those projects in the text IMO.

**Review by obnoxxx — CHANGES_REQUESTED** [review]

File: docs/release-process.md (hunk 2 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: fixed, thanks

File: docs/release-process.md (hunk 3 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: fixed, thanks

File: docs/release-process.md (hunk 4 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: fixed, thanks

File: docs/release-process.md (hunk 5 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: fixed, thanks

File: docs/release-process.md (hunk 6 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: ok, done

File: docs/release-process.md (hunk 7 above)
Comment (phlogistonjohn) [reply docs/release-process.md]: I merged it into a new sentance in the paragraph before the markdown block example. Does it seem better now?

File: docs/release-process.md (hunk 8 above)
Comment (phlogistonjohn) [reply docs/release-process.md:31]: ok, done.

**Review by obnoxxx — APPROVED** [review]
LGTM, thanks!

File: docs/release-process.md (hunk 7 above)
Comment (anoopcs9) [reply docs/release-process.md]: 👍🏻

**Review by anoopcs9 — APPROVED** [review]

**Pull request merged**